phpBB/phpbb/passwords/driver/argon2i.php

```diff
@@ -37,21 +37,12 @@ public function __construct(\phpbb\config\config $config, helper $helper, $memor
 	{
 		parent::__construct($config, $helper);
 
-		if ($this->is_sodium())
-		{
-			// For Sodium implementation, set special cost factor values (since PHP 7.4)
-			// See https://wiki.php.net/rfc/sodium.argon.hash and PHPBB3-16266
-			$this->memory_cost = max($memory_cost, 256*1024);
-			$this->threads     = 1;
-			$this->time_cost   = max($time_cost,   3);
-		}
-		else
-		{
-			// Otherwise don't allow cost factors to be below default settings
-			$this->memory_cost = max($memory_cost, 1024);
-			$this->threads     = max($threads,     2);
-			$this->time_cost   = max($time_cost,   2);
-		}
+		// For Sodium implementation, set special cost factor values (since PHP 7.4)
+		// See https://wiki.php.net/rfc/sodium.argon.hash and PHPBB3-16266
+		// Otherwise don't allow cost factors to be below default settings
+		$this->memory_cost = ($this->is_sodium()) ? max($memory_cost, 256*1024)	: max($memory_cost, 1024);
```

Comment:
Shouldn't it be possible to use the default constants that were introduced in PHP, e.g. `PASSWORD_ARGON2_DEFAULT_MEMORY_COST`? IIRC that shouldn't be an issue as this portion of the code shouldn't be exected if we're using an unsupported PHP version due to lazy loading? Could you check that maybe or should I do that?